tap_hubspot/__init__.py

```diff
@@ -307,8 +331,59 @@ def lift_properties_and_versions(record):
             record['properties_versions'] += versions
     return record
 
+def post_search_endpoint(url, data, params=None):
+
+    params, headers = get_params_and_headers(params)
+    headers['content-type'] = "application/json"
+
+    with metrics.http_request_timer(url) as timer:
+        resp = requests.post(
+            url=url,
+            json=data,
+            params=params,
+            headers=headers
+        )
+
+        resp.raise_for_status()
+
+    return resp
+
+def merge_responses(v1_data, v3_data):
+    for v1_record in v1_data:
+        v1_id = v1_record.get('dealId')
+        for v3_record in v3_data:
+            v3_id = v3_record.get('id')
+            if str(v1_id) == v3_id:
+                v1_record['properties'] = {**v1_record['properties'],
+                                           **v3_record['properties']}
+
+def process_v3_deals_records(v3_data):
+    """
+    This function:
+    1. filters out fields that don't contain 'hs_date_entered_*' and
+       'hs_date_exited_*'
+    2. changes a key value pair in `properties` to a key paired to an
+       object with a key 'value' and the original value
+    """
+    transformed_v3_data = []
+    for record in v3_data:
+        new_properties = {field_name : {'value': field_value}
+                          for field_name, field_value in record['properties'].items()
+                          if any(prefix in field_name for prefix in V3_PREFIXES)}
+        transformed_v3_data.append({**record, 'properties' : new_properties})
+    return transformed_v3_data
+
+def get_v3_deals(v3_fields, v1_data):
+    v1_ids = [{'id': str(record['dealId'])} for record in v1_data]
+
+    v3_body = {'inputs': v1_ids,
+               'properties': v3_fields,}
+    v3_url = get_url('deals_v3_batch_read')
+    v3_resp = post_search_endpoint(v3_url, v3_body)
+    return v3_resp.json()['results']
+
 #pylint: disable=line-too-long
```

Comment:
Not important for this PR but I normally add line-too-long to the disables in the pylint command. It's from a time when screens were not so large...